scripts/packager-codes/si-packagers-refresh.py

```diff
@@ -0,0 +1,214 @@
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+# PREREQUISITES
+python3
+apikey for geocode.maps.co (free account)
+
+# INSTALLATION
+## install virtual environment
+sudo apt install python3.11-venv
+python3 -m venv venv
+source venv/bin/activate
+
+## install needed packages
+pip install polars
+pip install requests
+
+# FETCH INPUT FILE
+# - from the Slovenian government website,
+#   download the actual list of "Seznam odobrenih živilskih obratov"
+#   (List of approved food establishments):
+#   https://www.gov.si/zbirke/storitve/odobritev-zivilskega-obrata/
+#   https://www.gov.si/assets/organi-v-sestavi/UVHVVR/Varna-hrana/Odobritev-obrata/Obrati-Zivila-O_ang.pdf
+# - download last version of https://github.com/tabulapdf/tabula-java/releases
+# - convert the pdf file into csv (update release number and file name):
+# $ java -jar tabula-1.0.5-jar-with-dependencies.jar   Obrati-Zivila-O_ang.pdf \
+#     --lattice --format CSV --pages all > slovenian_packaging_raw.csv
+
+# RUN
+update: api_key
+python3 si-packagers-refresh.py
+
+# POSTPROCESSING
+- deactivate the virtual environment:
+deactivate
+'''
+
+import polars as pl
+import re
+import requests
+import sys
+from time import sleep
+
+def clean_code(input_code: str) -> str:
+	# remove double spaces
+	input_code = input_code.replace('  ', ' ')
+
+	# SI H-1015 ES
+	if input_code.endswith('ES'):
+		input_code = input_code.replace('ES', '').strip()
+
+	# SI M-1035 SI
+	if input_code.endswith('SI'):
+		input_code = input_code.replace(' SI', '').strip()
+
+	# SI H-731, SI 731
+	if ',' in input_code:
+		input_code = "".join(input_code.split(', ')[1])
+
+	# SI H - 728, SI H 728, SI H-728, also with M
+	input_code = input_code.replace('H - ', 'H-')
+	input_code = input_code.replace('H ', 'H-')
+	input_code = input_code.replace('M - ', 'M-')
+	input_code = input_code.replace('M ', 'M-')
+
+	# SI - 907 -> SI 907
+	input_code = input_code.replace(' - ', ' ')
+	input_code = input_code.replace(' -', ' ')
+	input_code = input_code.replace('SI-', 'SI ')
+
+	# SI1194
+	if 'SI ' not in input_code:
+		input_code = input_code.replace('SI', 'SI ')
+	# SI M1106
+	if 'M-' not in input_code:
+		input_code = input_code.replace('M', 'M-')
+
+	return input_code
+
+
+def clean_address(input_address: str) -> str:
+	# special character because 
+	# sometimes new line between 2 addreses
+	# sometimes line for single address split in 2
+	input_address = "<>".join(input_address.split('\r'))
+
+	# fetch last occurence
+	# words 123A, place, 4567 city name
+	# Á found in a city name (PROSENJAKOVCI -PÁRTOSFALVA)
+	pattern = r'(([a-zčćžđšA-ZČĆŽĐŠŽ\s\-\.]+\d+[ABCDEFGIJ]?),(?:[a-zčćžđšA-ZČĆŽĐŠŽ\s\-\.\<\>]+,\s*)?[\<\>]*(\s*\d{4}[a-zčćžđšA-ZČĆŽĐŠŽÁ\s\-\.\<\>]+)$)'
```

Comment:
nit: Could you eventually use [verbose flag](https://docs.python.org/3/library/re.html#re.VERBOSE) and make it understandable ?

Also using named group for capture would be easier to understand.

Reply:
Also instead of `[a-zčćžđšA-ZČĆŽĐŠŽ\s\-\.]` why not use `[\w\s\-\.]`
that you can even write t`[\w\s.-]`
("." does not need to be escaped in intervals, as well as "-" if it is the last character)

Reply:
I tried using \w, but could not achieve same result. 
It may be due to \w including numbers. While I use \d{4} to recognize postal code.

For example, this input:
> KIDRIČEVA CESTA 63A, 4220 ŠKOFJA LOKA
> FUŽINSKA ULICA 1, 4220 ŠKOFJA LOKA"

Should lead to 
> FUŽINSKA ULICA 1, 4220 ŠKOFJA LOKA"